heartbeat/monitors/factory.go

```diff
@@ -40,15 +41,17 @@ import (
 // RunnerFactory that can be used to create cfg.Runner cast versions of Monitor
 // suitable for config reloading.
 type RunnerFactory struct {
-	info       beat.Info
-	addTask    scheduler.AddTask
-	byId       map[string]*Monitor
-	mtx        *sync.Mutex
-	pluginsReg *plugin.PluginsReg
-	logger     *logp.Logger
-	runOnce    bool
```

Comment:
runOnce is no longer needed since we determine the client type in heartbeat.go and pass in the right client creation function